omc3/scripts/kmod_summary.py

```diff
@@ -76,144 +99,235 @@ def output_kmod_summary_tables(
-            kmod_summaries[beam] = kmod_summary
-            tables_logbook[beam] = table_logbook
+                second_magnet = label.split("-")[1] # takes right magnet from magnet names. ex. [MQXA1.R5] from [MQXA1.L5-MQXA1.R5]
+                relevant = second_magnet.split(".")[-1] # takes last part of the second magnet name. ex. [R5] from [MQXA1.R5]
+                ip_number = relevant[1:] # isolate IP number. ex. [5] from [R5]
+                ip_name = f"{IP_COLUMN}{ip_number}"
+            df = data_file[COLS_X + COLS_Y].iloc[[0]] # returns a DataFrame with one row
+            df.insert(0, NAME_COLUMN, meas_name)
+            df.insert(0, IP_COLUMN, ip_name)
+            grouped.append(df)
+    return grouped
+
+def _collect_averaged_kmod_results(
+    averaged_meas: dict[str, dict[int, tfs.TfsDataFrame]],
+    beam: int
+    ) -> list[tfs.TfsDataFrame]:
 
-    if output_dir is not None:
-        for beam in beams:
-            _save_outputs(beam=beam, save_output_dir=output_dir, table_logbook=tables_logbook[beam], summary_df=kmod_summaries[beam])
-            if logbook:
-                _logbook_entry(logbook=logbook, beam=beam, save_output_dir=output_dir)
-        if not logbook:
-            LOG.info("Logbook name not provided: logbook entry not created.")
-    else:
-        LOG.info("Output_dir not provided: skipping file output for all beams.")
+    """
+    Gathers the various averaged Kmod results dataframes, taking only cols_x and cols_y values.
+
+    Args:
+        meas_paths (Sequence[Path | str]): List of measurement directories containing beam subfolders.
+        beam (int): Beam number to process.
+
+    Returns:
+        list[tfs.TfsDataFrame]: List containing grouped averaged kmod results.
+    """
+    LOG.info("Grouping averaged kmod results.")
+
+    grouped_averaged = []
+    for av_ip, av_res in averaged_meas.items():
+        LOG.info(f"Reading averaged results: {BEAM_DIR}{beam}, {av_ip}")
+        av_tab = av_res[0] # 0: averaged results table, 1, 2: are betx, bety for IP closest elements
+        av_tab_beam = av_tab.loc[[beam]]
+        df_averaged = av_tab_beam[COLS_X + COLS_Y]
+        df_averaged.insert(0, IP_COLUMN, av_ip)
+        grouped_averaged.append(df_averaged)
+    return grouped_averaged
+
+def _get_lumi_imbalance(
+    output_dir:Path | str
+    ) -> str:
+
+    """
+    Gathers the various luminosity imbalance dataframes in a single str, one line for each file.
+
+    Args:
+        output_dir (Path | str ): Path to the lumi imbalance files.
+    Returns:
+        str: str containing grouped luminosity imbalance results.
+    """
+    LOG.info("Getting luminosity imbalance values.")
+    output_dir = Path(output_dir)
+    prefix = EFFECTIVE_BETAS_FILENAME.split("{")[0] # if the file name starts with EFFECTIVE_BETAS_FILENAME till {ip}
+
+    report_lines = []
+    for file_path in output_dir.iterdir():
+        if not file_path.is_file() or not file_path.name.startswith(prefix):
```

Comment:
Probably rename `file_path` as it's not necessarily a file. Which makes the first `if` check necessary but strange to read